pkg/sdk/external_tables_integration_test.go

```diff
@@ -0,0 +1,397 @@
+				WithPartitionBy([]string{"filename"}).
+				WithDeltaTableFormat(Bool(true)).
+				WithAutoRefresh(Bool(false)).
+				WithRefreshOnCreate(Bool(false)).
+				WithCopyGrants(Bool(true)).
+				WithComment(String("some_comment")).
+				WithTag([]*TagAssociationRequest{NewTagAssociationRequest(tag.ID(), "tag-value")}),
+		)
+		require.NoError(t, err)
+
+		externalTable, err := client.ExternalTables.ShowByID(ctx, NewShowExternalTableByIDRequest(externalTableID))
+		require.NoError(t, err)
+		assert.Equal(t, externalTableID.Name(), externalTable.Name)
+	})
+
+	t.Run("Alter: refresh", func(t *testing.T) {
+		externalTableID := randomAccountObjectIdentifier(t)
+		err := client.ExternalTables.Create(ctx, minimalCreateExternalTableReq(externalTableID))
+		require.NoError(t, err)
+
+		err = client.ExternalTables.Alter(
+			ctx,
+			NewAlterExternalTableRequest(externalTableID).
+				WithIfExists(Bool(true)).
+				WithRefresh(NewRefreshExternalTableRequest("weather-nyc")),
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("Alter: add files", func(t *testing.T) {
+		externalTableID := randomAccountObjectIdentifier(t)
+		err := client.ExternalTables.Create(
+			ctx,
+			minimalCreateExternalTableReq(externalTableID).
+				WithPattern(String("weather-nyc/weather_2_3_0.json.gz")),
+		)
+		require.NoError(t, err)
+
+		err = client.ExternalTables.Alter(
+			ctx,
+			NewAlterExternalTableRequest(externalTableID).
+				WithIfExists(Bool(true)).
+				WithAddFiles([]*ExternalTableFileRequest{NewExternalTableFileRequest("weather-nyc/weather_0_0_0.json.gz")}),
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("Alter: remove files", func(t *testing.T) {
+		externalTableID := randomAccountObjectIdentifier(t)
+		err := client.ExternalTables.Create(
+			ctx,
+			minimalCreateExternalTableReq(externalTableID).
+				WithPattern(String("weather-nyc/weather_2_3_0.json.gz")),
+		)
+		require.NoError(t, err)
+
+		err = client.ExternalTables.Alter(
+			ctx,
+			NewAlterExternalTableRequest(externalTableID).
+				WithIfExists(Bool(true)).
+				WithAddFiles([]*ExternalTableFileRequest{NewExternalTableFileRequest("weather-nyc/weather_0_0_0.json.gz")}),
+		)
+		require.NoError(t, err)
+
+		err = client.ExternalTables.Alter(
+			ctx,
+			NewAlterExternalTableRequest(externalTableID).
+				WithIfExists(Bool(true)).
+				WithRemoveFiles([]*ExternalTableFileRequest{NewExternalTableFileRequest("weather-nyc/weather_0_0_0.json.gz")}),
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("Alter: set auto refresh", func(t *testing.T) {
+		externalTableID := randomAccountObjectIdentifier(t)
+		err := client.ExternalTables.Create(ctx, minimalCreateExternalTableReq(externalTableID))
+		require.NoError(t, err)
+
+		err = client.ExternalTables.Alter(
+			ctx,
+			NewAlterExternalTableRequest(externalTableID).
+				WithIfExists(Bool(true)).
+				WithAutoRefresh(Bool(true)),
+		)
+		require.NoError(t, err)
+	})
+
+	// TODO: Uncomment when the problem with alter external table set / unset tags is solved
```

Comment:
Add slack and jira link here pls, so that it's obvious from code what we are waiting for.

Reply:
Added Jira ticket id